src/LodeRunner.API.Test/IntegrationTests/ExecutingTestRun/TestRunExecution.cs

```diff
@@ -220,7 +229,9 @@ private async Task<(bool result, string fieldName, string conditionalValue)> Val
         private async Task<int> TryParseProcessOutputAndGetAPIListeningPort(List<string> outputList, int timeBetweenTriesMs = 1000, int maxRetries = 20)
         {
             int portNumber = 0;
-            for (int i = 1; i <= maxRetries; i++)
+
+            var taskSource = new CancellationTokenSource();
+            await Common.RunAndRetry(maxRetries, timeBetweenTriesMs, taskSource, async (int attemptCount) =>
             {
                 await Task.Delay(timeBetweenTriesMs).ConfigureAwait(false); // Log Interval is 5 secs.
 
```

Comment:
Is this Task.Delay still needed here? I see there is a Task.Delay in Common.RunAndRetry as well.

Reply:
No need for it, will remove